### PR TITLE
Enable pub publish dry run, set up dependabot

### PR DESCRIPTION
**Pull request by nohli**

### CODE REVIEW

**Comment by nohli** [comment]
Why is using `master` preferred?

File: .github/dependabot.yml
Comment (AlexV525) [inline .github/dependabot.yml:1]: Generally I'd prefer to update pub dependencies only when in major/minor updates, there should be no need to follow patches.

Comment (nohli) [reply .github/dependabot.yml:1]: This config is ignoring minor and patch updates for pub dependencies.

**Comment by AlexV525** [comment]
> Why is using `master` preferred?

This is suggested by the author in one of his comments, I cannot find it at present,

> I think, with the mock credentials, the actions doesn't work.

Actually this will work because if you're developing on forks, their repo would not contain an actual credential, and a dry-run does not require a valid credential.

File: .github/workflows/pub_dry_run.yml [hunk 1]
```diff
@@ -14,10 +14,10 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - name: Publish
+      - name: Dry run
         uses: k-paxian/dart-package-publisher@master
```
Comment (nohli) [inline .github/workflows/pub_dry_run.yml:18]:
```suggestion
        uses: k-paxian/dart-package-publisher@v1.6
```

File: .github/workflows/pub_publish.yml [hunk 2]
```diff
@@ -14,6 +14,6 @@ jobs:
       - name: Publish
         uses: k-paxian/dart-package-publisher@master
```
Comment (nohli) [inline .github/workflows/pub_publish.yml:15]:
```suggestion
        uses: k-paxian/dart-package-publisher@v1.6
```

File: .github/dependabot.yml
Comment (AlexV525) [reply .github/dependabot.yml:1]: Ah that's great

**Review by AlexV525 — APPROVED** [review]

**Pull request merged**